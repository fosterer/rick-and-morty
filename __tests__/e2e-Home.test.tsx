/**
 * @jest-environment jsdom
 */

import { screen, render, waitFor } from "@testing-library/react";
import Home from "@/pages/index";
import apolloClient from "@/lib/apollo";
import { ApolloProvider } from "@apollo/client";

test("Home renders inline with snapshot", async () => {
  const { container } = render(
    <ApolloProvider client={apolloClient}>
      <Home />
    </ApolloProvider>
  );

  await waitFor(() => screen.getByText("TEST Sanchez"));
  expect(container).toMatchSnapshot();
});
