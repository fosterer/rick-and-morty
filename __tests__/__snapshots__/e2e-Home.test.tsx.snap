// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders inline with snapshot 1`] = `
<div>
  <div
    class="relative"
  >
    <div
      class="fixed flex items-center z-10 w-full h-24 bg-slate-700"
    >
      <div
        class="flex flex-wrap items-center ml-3"
      >
        <div
          class="m-3 text-slate-100 text-lg"
        >
          Rick and Morty Series Explorer
        </div>
        <span>
          <input
            class="mx-5 ring px-2 py-1 rounded focus:outline-none focus:ring-blue-500 bg-slate-300 placeholder:text-slate-700"
            maxlength="20"
            placeholder="Type to discover..."
            type="text"
            value=""
          />
        </span>
      </div>
    </div>
    <div
      class="pt-24 z-0"
    >
      <div
        class="m-3 flex flex-wrap"
      >
        <div
          class="m-3 ring-2 rounded p-1 ring-yellow-800 bg-slate-400"
        >
          <ul
            class="text-center flex flex-col justify-center"
            style="width: 200px; height: 282px;"
          >
            <li
              class="animate-pulse"
            >
              Loading...
            </li>
          </ul>
        </div>
        <div
          class="m-3 ring-2 rounded p-1 ring-yellow-800 bg-slate-400"
        >
          <ul
            class="text-center flex flex-col justify-center"
            style="width: 200px; height: 282px;"
          >
            <li
              class="animate-pulse"
            >
              Loading...
            </li>
          </ul>
        </div>
        <div
          class="m-3 ring-2 rounded p-1 ring-yellow-800 bg-slate-400"
        >
          <ul
            class="text-center flex flex-col justify-center"
            style="width: 200px; height: 282px;"
          >
            <li
              class="animate-pulse"
            >
              Loading...
            </li>
          </ul>
        </div>
        <div
          class="m-3 ring-2 rounded p-1 ring-yellow-800 bg-slate-400"
        >
          <ul
            class="text-center flex flex-col justify-center"
            style="width: 200px; height: 282px;"
          >
            <li
              class="animate-pulse"
            >
              Loading...
            </li>
          </ul>
        </div>
        <div
          class="m-3 ring-2 rounded p-1 ring-yellow-800 bg-slate-400"
        >
          <ul
            class="text-center flex flex-col justify-center"
            style="width: 200px; height: 282px;"
          >
            <li
              class="animate-pulse"
            >
              Loading...
            </li>
          </ul>
        </div>
        <div
          class="m-3 ring-2 rounded p-1 ring-yellow-800 bg-slate-400"
        >
          <ul
            class="text-center flex flex-col justify-center"
            style="width: 200px; height: 282px;"
          >
            <li
              class="animate-pulse"
            >
              Loading...
            </li>
          </ul>
        </div>
        <div
          class="m-3 ring-2 rounded p-1 ring-yellow-800 bg-slate-400"
        >
          <ul
            class="text-center flex flex-col justify-center"
            style="width: 200px; height: 282px;"
          >
            <li
              class="animate-pulse"
            >
              Loading...
            </li>
          </ul>
        </div>
        <div
          class="m-3 ring-2 rounded p-1 ring-yellow-800 bg-slate-400"
        >
          <ul
            class="text-center flex flex-col justify-center"
            style="width: 200px; height: 282px;"
          >
            <li
              class="animate-pulse"
            >
              Loading...
            </li>
          </ul>
        </div>
        <div
          class="m-3 ring-2 rounded p-1 ring-yellow-800 bg-slate-400"
        >
          <ul
            class="text-center flex flex-col justify-center"
            style="width: 200px; height: 282px;"
          >
            <li
              class="animate-pulse"
            >
              Loading...
            </li>
          </ul>
        </div>
        <div
          class="m-3 ring-2 rounded p-1 ring-yellow-800 bg-slate-400"
        >
          <ul
            class="text-center flex flex-col justify-center"
            style="width: 200px; height: 282px;"
          >
            <li
              class="animate-pulse"
            >
              Loading...
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</div>
`;
